src/audio/crossover/crossover.c

```diff
@@ -703,16 +699,17 @@ static int crossover_copy(struct comp_dev *dev)
 	}
 
 	/* Process crossover */
-	buffer_invalidate(source, source_bytes);
 	cd->crossover_process(dev, source, sinks, num_sinks, frames);
+	source = buffer_release(source);
+	comp_update_buffer_consume(source, source_bytes);
 
 	for (i = 0; i < num_sinks; i++) {
 		if (!sinks[i])
 			continue;
 		buffer_writeback(sinks[i], sinks_bytes[i]);
+		sinks[i] = buffer_release(sinks[i]);
```

Comment:
it's rather unclear to me when a writeback is explicitly required and when it's handled by the release?